.github/workflows/ci.yml

```diff
@@ -0,0 +1,164 @@
+          echo ::set-output name=repo_name::${REPO_NAME}
+      ## This clones and checks out.
+      - name: Checkout branch from Github
+        uses: actions/checkout@v2
+      ## Setup node and npm caching.
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.13.0'
+          cache: 'npm'
+          cache-dependency-path: '**/package-lock.json'
+      - name: Install NPM 8
+        run: |
+          npm install -g npm@8.1.0
+      ## Setup PHP and Composer Caching
+      - name: Setup PHP with tools
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
+          ## This list of extensions matches those we install with our docker file.
+          extensions: bz2, cli, common, curl, fpm, gd, json, mbstring, memcached, mysql, oauth, opcache, readline, sqlite3, soap, xdebug, xml
+          tools: composer:v2
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        run: |
+          echo "::set-output name=dir::$(composer config cache-files-dir)"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+      - name: Composer install
+        run: |
+          composer install
+      ## Install BLT and Drush Launchers
+      - name: Install BLT and Drush Launchers
+        run: |
+          curl -sL -o /usr/local/bin/drush https://github.com/drush-ops/drush-launcher/releases/download/0.6.0/drush.phar && chmod +x /usr/local/bin/drush
+          curl -sL -o /usr/local/bin/blt https://github.com/acquia/blt-launcher/releases/download/v1.0.3/blt.phar && chmod +x /usr/local/bin/blt
+      ## Create MySQL databases
+      - name: Turn on MySQL and Create DBs
+        run: |
+          sudo /etc/init.d/mysql start
+      - name: Setup Env
+        run: |
+          ## The setup env script does a few things we cannot.
+          mysql -u root -proot -e "CREATE DATABASE IF NOT EXISTS drupal"
+          mysql -u root -proot -e "GRANT ALL ON drupal.* to 'drupal'@'%' IDENTIFIED BY 'drupal'"
+          mysql -u root -proot -e "CREATE DATABASE IF NOT EXISTS simpletest"
+          mysql -u root -proot -e "GRANT ALL ON simpletest.* to 'simpletest'@'%' IDENTIFIED BY 'simpletest'"
+          ## We don't set git info for right now as we are in the NCIOCPL origin right now.
+      - name: Validate
+        run: |
+          blt validate --define drush.alias='${drush.aliases.ci}' --ansi --verbose --no-interaction
+      - name: Setup App
+        run: |
+          ## The setup app script cannot create build.json as we are in the NCIOCPL origin.
+          export PHANTOMJS_CDNURL=https://npm.taobao.org/mirrors/phantomjs
+          blt setup --define drush.alias='${drush.aliases.ci}' --environment=ci --no-interaction --ansi --verbose
+      - name: Test
```

Comment:
Splitting these apart into individual test steps could help (slightly) in tracking them.

Reply:
I split apart the tests into Behat and PHPUnit.